client/src/shared/state/Inventory.js

```diff
@@ -104,16 +110,76 @@ class Inventory {
         if (!ItemTypes[itemConfig.typeCode].hasUseEffect) return;
 
         this.hotbar.push(itemConfig);
-
+        this.saveItemId[itemConfig.id] = itemConfig.slotIndex;
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
     }
 
     removeFromHotbar(itemConfig) {
+        delete this.saveItemId[itemConfig.id];
         this.hotbar = this.hotbar.filter((eachItem) => eachItem !== itemConfig);
-
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
+    }
+
+    saveHotbar() {
+        window.localStorage.setItem(this.loadHotBarRequest, JSON.stringify(this.saveItemId));
+    }
+
+    defaultHotBar() {
+        this.items.forEach((itemConfig) => {
+            this.addToHotbar(itemConfig);
+        });
+    }
+
+    loadFromLocalStorage(playerID) {
+        this.saveSession = true;
+        this.userName = playerID;
+        const gameName = "dungeonz";
+        const requestHotbar = "hotBar";
+        this.loadHotBarRequest = gameName + this.userName + requestHotbar;
+    }
+
+    loadHotbar(playerID, isLoggedIn) {
+        if (isLoggedIn === true) {
+            this.loadFromLocalStorage(playerID);
+        }
+        const loadStorageHotbar = window.localStorage.getItem(this.loadHotBarRequest);
+        if (this.saveSession === false || loadStorageHotbar === null) {
+            this.defaultHotBar();
+            return;
+        }
+        this.saveItemId = JSON.parse(loadStorageHotbar);
+        this.populateInvetorytoHotbar();
+    }
+
+    populateInvetorytoHotbar() {
+        Object.keys(this.saveItemId).forEach((key, value) => {
```

Comment:
For these loops, `(key, value)` isn't particularly accurate, as the parameters passed into the forEach callback are `(element, index)`.
Where element is the value of the current array element, which is more descriptive.
Though in this context that value is literally the name of a key... Humm...
It could be `savedItemId` instead, as using `(key, value)` or `(element, index)` etc. for loops is just a generic way of describing the properties, but it is also common to use more descriptive words for the property for what the value actually represents.
In either case, the second property (`value`) isn't being used, so can be removed.
i.e.
`Object.keys(this.saveItemId).forEach((savedItemId) => { `